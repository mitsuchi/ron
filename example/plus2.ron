op 50 : S _
op 50 : _ + _ = _

0 + n = n
(S n) + m = (S l) {
    n + m = l
}

main {
    S 0 + S S 0 = x
}