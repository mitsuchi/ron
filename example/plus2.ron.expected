x = S S S 0
